### PR TITLE
Ripley mechs (and other mechs that are missing an air tank) can no longer toggle the air source.

### LINKED ISSUE #66005: Ripley MK-I erroneously allows toggling internal air tank in the UI.
## Testmerges:
#65589: 'Biddle Verbs: Queues the Most Expensive Verbs for the Next Tick if the Server Is Overloaded' by Kylerace at commit 70246ca2f5
#64124: 'Refactor and improve antimagic to be more robust' by timothymtorres at commit 9fde5f7c4a
#65943: 'tgui: Improvements and documentation of low level APIs' by stylemistake at commit 194dfd7dc0

## Reproduction:
- Enter MK-I and open mech UI (View Stats).
- Observe clickable UI button next to "Air Source".
- Click button to toggle to internal tank.
- Suffocate and possibly die because there is no internal tank on the MK-I.
- Alternatively, be immune to fires in the surrounding enviroment if you wear an EVA suit

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
Disables the button on the UI if the mech has no internal tank. Also early returns on procs if the UI somehow lets the click through.

Fixes #66005

## Why It's Good For The Game
Bugfix.

## Changelog

:cl:
fix: Ripley mechs can no longer toggle to their nonexistant air tank.
/:cl:

### CODE REVIEW

**Review by TiviPlus — APPROVED** [review]

File: code/modules/vehicles/mecha/mecha_actions.dm [hunk 1]
```diff
@@ -34,6 +34,12 @@
 	if(!owner || !chassis || !(owner in chassis.occupants))
 		return
 
+	if(!chassis.internal_tank) //Just in case.
+		chassis.use_internal_tank = FALSE
+		chassis.balloon_alert(owner, "No tank available!")
```
Comment (Fikou) [inline code/modules/vehicles/mecha/mecha_actions.dm:39]: balloon alerts are not supposed to be capitalized

Comment (zxaber) [reply code/modules/vehicles/mecha/mecha_actions.dm]: Changed.

File: code/modules/vehicles/mecha/mecha_ui.dm [hunk 2]
```diff
@@ -115,6 +115,7 @@
 	data["power_max"] = cell?.maxcharge
 	data["mecha_flags"] = mecha_flags
 	data["internal_damage"] = internal_damage
+	data["airtank_present"] = internal_tank ? TRUE : FALSE
```
Comment (MrMelbert) [inline code/modules/vehicles/mecha/mecha_ui.dm:118]:
```suggestion
	data["airtank_present"] = !!internal_tank
```
as an option, you can cast to boolean instead

File: tgui/packages/tgui/interfaces/Mecha/data.ts [hunk 3]
```diff
@@ -69,6 +69,7 @@ export type OperatorData = {
   mecha_flags: number;
   internal_damage: number;
   internal_damage_keys: string[];
+  airtank_present: boolean;
```
Comment (Watermelon914) [inline tgui/packages/tgui/interfaces/Mecha/data.ts:72]: this should be BooleanLike
Booleans on byond are numbers, not actual bools

**Comment by tralezab** [comment]
conflict

**Pull request merged**